src/main.js

```diff
@@ -1,96 +1,151 @@
-import logger from '@financial-times/n-logger';
-
-const MASK_SEQUENCE = '*****';
-
-export default class {
-
-	constructor (sensitiveFields = []) {
-		const fieldsToMask = sensitiveFields.join('|');
-		this.sensitiveFields = new RegExp(`(${fieldsToMask})[\\\s]*\\=[\\\s]*[\\\]?[\\"\\']?([\\S]+)[\\\]?[\\"\\']?|(${fieldsToMask})`, 'ig');
+const logger = require('@financial-times/n-logger').default;
+
+/**
+ * MaskLogger Class
+ * @example
+ * const logger = new MaskLogger(['denyField'], ['allowField'], '*MASKED*');
+ * logger.debug('Data which may contain sensitive information', data);
+ */
+class MaskLogger {
+	/**
+	 * Create a MaskLogger
+	 * @param {array<string>} denyList Extra field names to mask
+	 * @param {array<string>} allowList Field names to allow from the default deny list
+	 * @param {string?} maskString Mask to be applied to sensitive values
+	 */
+	constructor (denyList = [], allowList = [], maskString = '*****') {
+		this.maskString = maskString;
+		this.maskList = [
+			'email',
+			'password',
+			'name',
+			'firstName',
+			'lastName',
+			'phone',
+			'primaryTelephone',
+			'postcode',
+			'session',
+			...denyList
+		];
+
+		// Deduplicate the maskList to increase masking performance
+		this.maskList = [...new Set(this.maskList)];
```

Comment:
Oooo. I like this!

Question, do you know if computationally is this more efficient than using a filter() or reduce() to deduplicate an array? (Curious if it would be a consideration when working with a bigger array).

Reply:
Well... looks like this this way is consistently slower than filter! http://jsben.ch/sWq0A

Reply:
Good to know. And thanks for sharing the link - not seen js benchmark before!